Rubberduck.Main/Root/RubberduckIoCInstaller.cs

```diff
@@ -810,22 +808,21 @@ private void RegisterCustomDeclarationLoadersToParser(IWindsorContainer containe
 
         public static IEnumerable<Assembly> AssembliesToRegister()
         {
-            var assemblies = new[]
-                {
-                    Assembly.GetExecutingAssembly(),
-                    Assembly.GetAssembly(typeof(IHostApplication)),
-                    Assembly.GetAssembly(typeof(IParseCoordinator)),
-                    Assembly.GetAssembly(typeof(IIndenter))
-                }
-                .Concat(FindPlugins());
-            return assemblies;
+            return Assembly.GetExecutingAssembly()
+                .GetReferencedAssemblies()
+                .Where(name => name.FullName.StartsWith("Rubberduck"))
```

Comment:
Why do we use a name based criterion here instead of hardcoding the assemblies or determining them by being referenced? I see a potential security concern here. (We run all inspections from these assemblies.)

Reply:
this is just an additional restriction on the ReferencedAssemblies. Do note that the arbitrary code execution is not in that place, but about five lines down in `FindPlugins`:

https://github.com/Vogel612/Rubberduck/blob/6df1c57bb2daa2defab2276bb637eddf90706ab9/Rubberduck.Main/Root/RubberduckIoCInstaller.cs#L838